src/init.c

```diff
@@ -65,18 +65,6 @@ void end_sigill_section(struct sigaction *oldact, sigset_t *oldset)
 	sigprocmask(SIG_SETMASK, oldset, NULL);
 }
 
-#if OPENSSL_VERSION_PREREQ(3, 0)
-static void openssl_cleanup()
```

Comment:
How about making this a public API function and document that the user should call this before dlclose()?  That seems to be the cleanest solution.  Would also be an API addition (thus a minor release).

Reply:
Would be OK for me. We should probably use a different name then, like ica_openssl_cleanup().....
And we will need to have this function independent on the OpenSSL version. For < OpenSS 3.0 this would just be a no-op.

Not sure if users will really call that function then, but at least there is a possibility to do it right then.

@jschmidb What do you think?

Reply:
That would certainly be ok with me. Would opencryptoki then use this new API function?

Reply:
Well, this would then mean a version dependency... not sure how to deal with tis best?
So I guess sooner or later it would (as well as the IBMCA engine I guess)

Reply:
For the IBMCA engine, that is easy.  Lookup the symbol via dlsym.  If it is there, call it before dlclose.  If not, do not call it.  For the provider, this might be trickier.  But that is what autoconf is for...